### PR TITLE
[style] Sort stops dictionaries before conversion to expression

### PR DESCRIPTION
**Pull request by nishant-karajgikar**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.

 - [x] Write tests for all new functionality. If tests were not written, please explain why.

 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog>Dictionaries passed to expressions are now sorted by default</changelog>`.

### Summary of changes
This PR fixes a bug where unsorted stops dictionaries would not be processed by the style renderer. The fix essentially makes it so that the ExpressionBuilder sorts the stops dictionary (by key) before generating the expression elements.

### CODE REVIEW

File: Mapbox/MapboxMapsStyleTests/ExpressionTests/ExpressionBuilderTests.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import XCTest
+#if canImport(MapboxMaps)
+@testable import MapboxMaps
+#else
+@testable import MapboxMapsStyle
+#endif
+
+internal class ExpressionBuilderTests: XCTestCase {
```
Comment (neelmistry94) [inline Mapbox/MapboxMapsStyleTests/ExpressionTests/ExpressionBuilderTests.swift:8]: Would be good to add a test that checks for a negative key to make sure potential future changes doesn't cause this bug

Comment (nishant-karajgikar) [reply Mapbox/MapboxMapsStyleTests/ExpressionTests/ExpressionBuilderTests.swift:8]: Hmm can't test for that since there's an `assertionFailure` unfortunately. The original bug was also actually caused by an unsorted stops dictionary. That's what's being tested below.

**Review by neelmistry94 — APPROVED** [review]
I tested your branch locally with respect to the issue that was opened and it LGTM. 

I have limited expression experience so you might want to get a second pair of eyes to look at the code just in case.
